scripts/zones/Windurst_Waters/npcs/Moari-Kaaori.lua

```diff
@@ -73,36 +73,41 @@ function onEventFinish(player,csid,option)
 
     if (csid == 514 and option == 1) then
         if (player:getQuestStatus(WINDURST,SAY_IT_WITH_FLOWERS) == QUEST_COMPLETED) then
-            player:setVar("FLOWER_PROGRESS",2);
+            player:setVar("FLOWER_PROGRESS",1);
         else
             player:addQuest(WINDURST,SAY_IT_WITH_FLOWERS);
+            player:setVar("FLOWER_PROGRESS",1);
         end
     elseif (csid == 520) then -- First completion, Iron Sword awarded.
-        player:tradeComplete();
-        player:completeQuest(WINDURST,SAY_IT_WITH_FLOWERS);
-        player:addFame(WINDURST,30);
-        player:addItem(16536);
-        player:messageSpecial(ITEM_OBTAINED,16536);
-        player:setVar("FLOWER_PROGRESS",0);
-        player:setLocalVar("FLOWER_ZONE",1);
-        player:setTitle(CUPIDS_FLORIST);
+        if (player:getFreeSlotsCount() > 0) then
+            player:tradeComplete();
+            player:addItem(16536);
+            player:completeQuest(WINDURST,SAY_IT_WITH_FLOWERS);
+            player:addFame(WINDURST,30);
+            player:messageSpecial(ITEM_OBTAINED,16536);
+            player:setVar("FLOWER_PROGRESS",0);
+            player:needToZone(true);
+            player:setTitle(CUPIDS_FLORIST);
+        else
+            player:messageSpecial(ITEM_CANNOT_BE_OBTAINED,16536);
+        end
     elseif (csid == 522) then -- Wrong flowers so complete quest, but smaller reward/fame and no title.
         player:completeQuest(WINDURST,SAY_IT_WITH_FLOWERS);
         player:tradeComplete();
-        player:addGil(100);
+        player:addGil(GIL_RATE * 100);
         player:messageSpecial(GIL_OBTAINED,100);
         player:addFame(WINDURST,10);
-        player:setLocalVar("FLOWER_ZONE",1);
+        player:needToZone(true);
         player:setVar("FLOWER_PROGRESS",0);
     elseif (csid == 525) then -- Repeatable quest rewards.
         player:tradeComplete();
         player:addFame(WINDURST,30);
-        player:addGil(GIL_RATE*400);
+        player:addGil(GIL_RATE * 400);
         player:setVar("FLOWER_PROGRESS",0);
-        player:setLocalVar("FLOWER_ZONE",1);
+        player:needToZone(true);
```

Comment:
can also table this

Reply:
hmm, not quite sure what to table on

Reply:
ah i misread the code above, should be fine